### ISSUE TITLE
Installation Problems

### ISSUE THREAD

**Issue by Olfried2** [sreimers/zoom-l8#2]
Hello,
thanks for the Zoom-driver! but it doesn't install on my Ubuntu 22.04.1 LTS. 
As the way you described to install failed, (with no response)
I changed to /usr/src/snd-usb-zoom-0.0.1 and called "make" and got the following informations

sudo make
make -C /lib/modules/`uname -r`/build M= modules
make[1]: Entering directory '/usr/src/linux-headers-5.15.0-58-generic'
  SYNC    include/config/auto.conf.cmd
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  HOSTCC  scripts/kconfig/confdata.o
  HOSTCC  scripts/kconfig/expr.o
  LEX     scripts/kconfig/lexer.lex.c
/bin/sh: 1: flex: not found
make[3]: *** [scripts/Makefile.host:9: scripts/kconfig/lexer.lex.c] Error 127
make[2]: *** [Makefile:622: syncconfig] Error 2
make[1]: *** [Makefile:742: include/config/auto.conf.cmd] Error 2
make[1]: *** [include/config/auto.conf.cmd] Deleting file 'include/generated/autoconf.h'
make[1]: Leaving directory '/usr/src/linux-headers-5.15.0-58-generic'
make: *** [Makefile:11: build] Error 2

Could you help with this ?
Thanks in advance

**Comment by Olfried2** [comment]
After some investigation I realized that I have to install **flex** another _make_ shows that there has to be **bison** also installed.

A new error appears
`
No rule to make target  arch/x86/entry/syscalls/syscall_32.tbl, 
needed by ``arch/x86/entry/syscalls/../../include/generated/asm/syscalls_32.h''
`
Another googling suggests with this error to install the linux headers
`sudo apt-get install linux-headers-...` 
Replace the ... with your kernel version - for me it was linux-headers-**5.15.0-58**-generic. I didn't realized the dot against number stuff so it took more than two hours to download, unzip and install (all available headers).
`sudo make clean`
followed by `sudo make` didn't solve the problem only the error is different.
<code>
make -C /lib/modules/`uname -r`/build M= modules
make[1]: Entering directory '/usr/src/linux-headers-5.15.0-58-generic'
make[2]: *** No rule to make target 'arch/x86/entry/syscalls/syscall_32.tbl', needed by 'arch/x86/include/generated/uapi/asm/unistd_32.h'.  Stop.
make[1]: *** [arch/x86/Makefile:217: archheaders] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.15.0-58-generic'
make: *** [Makefile:11: build] Error 2
</code>

Maybe it's not important, but my first _make_ made as normal user threw this errors
<code>
make -C /lib/modules/`uname -r`/build M=/usr/src/snd-usb-zoom-0.0.1 modules
make[1]: Entering directory '/usr/src/linux-headers-5.15.0-58-generic'
mkdir: cannot create directory ‘/usr/src/snd-usb-zoom-0.0.1/.tmp_3257920’: Permission denied
...
arch/x86/Makefile:142: CONFIG_X86_X32 enabled but no binutils support
</code>
_binutils is already the newest version (2.38-4ubuntu2.1)_

I am **baffled**. Defintly help is required.

**Comment by sreimers** [comment]
You can use `cd /usr/src/snd-usb-zoom-0.0.1 && sudo make build` for manual build, but I just tested the dkms way from the readme on a fresh virtual machine and it should work too.

**Comment by Olfried2** [comment]
Thanks for the hint.
I edited my previous comment - I had to install **linux-headers-5.15.0-58-generic**.

So I tested _make build_ with errors
<code>
make -C /lib/modules/`uname -r`/build M= modules
make[1]: Entering directory '/usr/src/linux-headers-5.15.0-58-generic'
make[2]: *** No rule to make target 'arch/x86/entry/syscalls/syscall_32.tbl', needed by 'arch/x86/include/generated/uapi/asm/unistd_32.h'.  Stop.
make[1]: *** [arch/x86/Makefile:217: archheaders] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.15.0-58-generic'
make: *** [Makefile:11: build] Error 2
</code>
After a _make clean_ I tested 
<code>
sudo dkms autoinstall
</code>
It worked in the background but didn't talk to me. After a reboot I ran _alsa-info_ which gave me the already known sound devices only.

**But** _sudo dmesg_ gave me a new information

<code>
[ 1833.278005] usb 1-2.3: new high-speed USB device number 12 using xhci_hcd
[ 1833.480503] usb 1-2.3: New USB device found, idVendor=1686, idProduct=0515, bcdDevice= 0.06
[ 1833.480515] usb 1-2.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1833.480521] usb 1-2.3: Product: L-8
[ 1833.480525] usb 1-2.3: Manufacturer: ZOOM Corporation
[ 1833.480529] usb 1-2.3: SerialNumber: 70960464094E00000000000000000000
</code>
The driver seems to be working. Now the alsa-configuration-nightmare begins. 

I'd like to add my steps tomorrow (If I get it configured and running) for further reference.

**Comment by Olfried2** [comment]
Ok, I deleted my third attempt to install the zoom driver because while I was preparing the stuff, an update to a newer kernel was done in the background (which I didn't realize).

BTW. the `sudo dkms autoinstall` didn't work either.

I had to install the modules for the actual kernel _/lib/modules/5.15.0-60-generic_.
Used `depmode` to add them to the valid libraries
Then I had to `make build` the sourcecode in the /usr/src/snd-usb-zoom-0.0.1 directory. 
_If there is the file snd-usb-zoom.ko in the directory after compiling, everything is fine_
After that  `sudo modprobe snd-usb-zoom` - add the module to the kernel.
To make it persistent  `sudo echo snd-usb-zoom | sudo tee -a /etc/modules`.
now `lsmod | grep zoom` gives an reliable output (even after reboot)
<code bash>
snd_usb_zoom           24576  0
snd_pcm               143360  9 snd_usb_zoom,snd_hda_codec,soundwire_intel,snd_sof,snd_sof_intel_hda_common,snd_compress,snd_soc_core,snd_hda_core,snd_pcm_dmaengine
snd                   106496  10 snd_seq,snd_seq_device,snd_usb_zoom,snd_hwdep,snd_hda_codec,snd_timer,snd_compress,snd_soc_core,snd_pcm,snd_rawmidi
</code>

Next will be a trial to add the _snd_pcm_  or _snd_usb_zoom_ to the alsa universe.
as in _/usr/share/alsa/alsa.conf_ noted, several possibilities are given (or maybe /etc/asound.conf).
<code>
"/var/lib/alsa/conf.d"
			"/usr/etc/alsa/conf.d"
			"/etc/alsa/conf.d"
			"/etc/asound.conf|||/usr/etc/asound.conf"
			"~/.asoundrc"
</code>
and further
<code>
"/cards/aliases.conf"
"/var/lib/alsa/card"
</code>

As the zoom is not recognized by _alsa_ yet some assistance would be helpful.

**Comment by Olfried2** [comment]
After spending now more than 30 hours to get it working, I would like to ask kindly if You can confirm to have a _working_ system.
Actually the zoom will be recognized by lsusb and the driver shows up in lsmod. Even in VMWare (running Win10) the zoom is present. But any soundsystem will not work with it. 
If you have a working system, could you please provide type, version of your linux distribution and the used sound system (alsa, pulseaudio, jack....)?
Thanks in advance

**Comment by Olfried2** [comment]
After spending now more than 30 hours to get it working, I would like to ask kindly if You can confirm to have a _working_ system.
Actually the zoom will be recognized by lsusb and the driver shows up in lsmod. Even in VMWare (running Win10) the zoom is present. But any soundsystem will not work with it. 
If you have a working system, could you please provide type, version of your linux distribution and the used sound system (alsa, pulseaudio, jack....)?
Thanks in advance

**Comment by sreimers** [comment]
Yes its working on a current Arch Linux. Your USB `idVendor=1686, idProduct=0515` looks wrong. idProduct should be `0525`, which USB mode is configured on the Zoom-L8?

**Comment by sreimers** [comment]
Looks like you are using 44,1kHz, can you switch to 48kHz in device settings (`System > Sample Rate`)?

**Comment by Olfried2** [comment]
Another 16hours waste of time ?
Yo, the sample rate setting was wrong _[ You have to adjust to 48kHz (System -> sample rate) and then shut off using the switch on the back. Otherwise if you just disconnect the USB power the setting will be lost and you start on 44.1kHz at next bootup]_.

The L8 will be shown, and I can choose channel 1 and 2 for recording. Other channel > 2 have to be additional selected. My DAW is Harris Mixbus8 /Ardour. In Audacity only "ZOOM L-8:USB Audio (hw:0,0)" is found in recording as in playback. Recording works - playback not (same hw address) 

Alsa reports that this device does not have any controls. 

I set up a laptop with Ubuntu 22.04.2 LTS in minimal installation to prevent any disturbance. I actually installed Audacity to test the possibilities. Where I also get the already described information.

The USBcable is good quality with added ferrite and I tried it also with an external powered hub.

If you have no other hints for getting it work, my last attempt will be installing Arch, could you provide which desktop you're using and which soundsystem alsa, pulsaudio..?

Thanks in advance
